client/src/pages/LyricsView/LyricsDisplay.tsx

```diff
@@ -52,50 +78,116 @@ const LyricsDisplay: React.FunctionComponent<IProps> = ({
+        const currObject = { array: progressArray, before, highlighted, after };
+        return currObject;
+      });
+    }
+  }, [syncedLyricsArray, progressMs]);
+
   const onUserSearch = (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) =>
     setSearch(event.currentTarget.value === undefined ? "" : event.currentTarget.value);
   const toggleSearchShown = () => setSearchShown(s => !s);
+  const toggleSyncEnabled = () => {
+    setSyncEnabled(s => !s);
+  };
 
   return (
     <div className={classes.root}>
-      {/* TODO https://material-ui.com/components/text-fields/#input-adornments */}
-      {searchShown ? (
-        <Box mb={1}>
-          <TextField
-            variant="outlined"
-            value={search}
-            onChange={onUserSearch}
-            InputProps={{
-              endAdornment: (
-                <InputAdornment position="end">
-                  <IconButton onClick={toggleSearchShown} edge="end">
-                    <CloseIcon />
-                  </IconButton>
-                </InputAdornment>
-              )
-            }}
-            label="Search"
-            placeholder="Search lyrics you heard to find your position..."
-            style={{ width: "100%", maxWidth: 600 }}
-          />
-        </Box>
-      ) : (
-        <IconButton className={classes.toggleSearchButton} onClick={toggleSearchShown}>
-          <SearchIcon fontSize="small" />
-        </IconButton>
-      )}
+      <Toolbar className={classes.toolbar}>
+        {/* TODO https://material-ui.com/components/text-fields/#input-adornments */}
+        {searchShown ? (
+          <Box mb={1}>
+            <TextField
+              variant="outlined"
+              value={search}
+              onChange={onUserSearch}
+              InputProps={{
+                endAdornment: (
+                  <InputAdornment position="end">
+                    <IconButton onClick={toggleSearchShown} edge="end">
+                      <CloseIcon />
+                    </IconButton>
+                  </InputAdornment>
+                )
+              }}
+              label="Search"
+              placeholder="Search lyrics you heard to find your position..."
+              style={{ width: "100%", maxWidth: 600 }}
+            />
+          </Box>
+        ) : (
+          <IconButton onClick={toggleSearchShown}>
+            <SearchIcon fontSize="small" />
+          </IconButton>
+        )}
+        {syncPossible ? (
+          <IconButton onClick={toggleSyncEnabled}>
+            {syncEnabled ? <SyncEnabledIcon /> : <SyncDisabledIcon />}
+          </IconButton>
+        ) : (
+          <SyncDisabledIcon />
+        )}
+      </Toolbar>
       {lyrics ? (
         <div>
           <Typography component="div" className={classes.lyrics} ref={lyricsRef}>
-            {lyrics}
+            {syncedLyrics.before}
+            <br />
+            <span id="highlighted-text" className={classes.mark} ref={highlightedRef}>
```

Comment:
You can drop the id since you're not using it now

```suggestion
            <span className={classes.mark} ref={highlightedRef}>
```

Great job on using the ref btw.